src/xunit.runner.uap/RunnerRemoteExecutionServiceUAP/RunnerRemoteExecutionServiceUAP.cs

```diff
@@ -0,0 +1,146 @@
+            HandleTheRequest(args.Request.Message, returnData);
+
+            // Return the data to the caller.
+            // Complete the deferral so that the platform knows that we're done responding to the app service call.
+            // Note for error handling: this must be called even if SendResponseAsync() throws an exception.
+            await args.Request.SendResponseAsync(returnData); 
+
+            messageDeferral.Complete();
+        }
+
+        private void OnTaskCanceled(IBackgroundTaskInstance sender, BackgroundTaskCancellationReason reason)
+        {
+            if (this.backgroundTaskDeferral != null)
+            {
+                // Complete the service deferral.
+                this.backgroundTaskDeferral.Complete();
+            }
+        }
+
+        private void HandleTheRequest(ValueSet message, ValueSet returnData)
+        {
+            string assemblyName;
+            string typeName;
+            string methodName;
+            string[] additionalArgs;
+
+            // default the results to 0 as success. We override it at failures.
+            returnData["Results"] = 0;
+
+            // The message expects to be passed the target assembly name to load, the type
+            // from that assembly to find, and the method from that assembly to invoke.
+            // Any additional arguments are passed as strings to the method.
+
+            if (!GetStringValueFromValueSet(message, "AssemblyName", out assemblyName) || String.IsNullOrEmpty(assemblyName))
+            {
+                returnData["Results"] = -1;
+                returnData["Log"] = $"RemoteExecuter Error: Missing assembly name";
+                return;
+            }
+
+            if (!GetStringValueFromValueSet(message, "TypeName", out typeName) || String.IsNullOrEmpty(typeName))
+            {
+                returnData["Results"] = -2;
+                returnData["Log"] = $"RemoteExecuter Error: Missing type name";
+                return;
+            }
+
+            if (!GetStringValueFromValueSet(message, "MethodName", out methodName) || String.IsNullOrEmpty(methodName))
+            {
+                returnData["Results"] = -3;
+                returnData["Log"] = $"RemoteExecuter Error: Missing method name";
+                return;
+            }
+
+            List<string> argList = new List<string>();
+            int i = 0;
+            string arg;
+            while (GetStringValueFromValueSet(message, "Arg"+i, out arg))
+            {
+                argList.Add(arg);
+                i++;
+            }
+            additionalArgs = argList.ToArray();
+
+            // Load the specified assembly, type, and method, then invoke the method.
+            // The program's exit code is the return value of the invoked method.
+            object instance = null;
+            string log = $"RemoteExecuter: {assemblyName} {methodName} {string.Join(", ", additionalArgs)}{Environment.NewLine}";
```

Comment:
Should we change this to a StringBuilder and use it to log more info from this code? Would that help debug this in the future?